echopop/inversion/inversion_matrix.py

```diff
@@ -0,0 +1,1230 @@
+    the most promising realization.
+
+    Parameter scaling normalizes all parameters to [0,1] range, which
+    improves numerical conditioning when parameters have very different
+    scales (e.g., length in mm vs density in kg/m³).
+
+    The class supports various scattering models through a plugin architecture
+    defined in `SCATTERING_MODEL_PARAMETERS`. Currently implemented models
+    include PCDWBA for elongated organisms.
+
+    Raises
+    ------
+    EchopopValidationError
+        If data or simulation_settings fail validation requirements
+    """
+
+    def __new__(
+        cls,
+        data: pd.DataFrame,
+        simulation_settings: Dict[str, Any],
+        verbose: bool = True,
+    ):
+        # Validate
+        try:
+            # ---- Check
+            valid_args = ValidateInversionMatrix.create(
+                **dict(data=data, simulation_settings=simulation_settings)
+            )
+        # Break creation
+        except ValidationError as e:
+            raise EchopopValidationError(str(e)) from None
+
+        # Create instance
+        self = super().__new__(cls)
+
+        # Update attributes
+        self.measurements = valid_args["data"].copy()
+        self.simulation_settings = valid_args["simulation_settings"]
+        self.verbose = verbose
+
+        # Generate
+        return self
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        simulation_settings: Dict[str, Any],
+        verbose: bool = True,
+    ):
+        """
+        Initialize the InversionMatrix with acoustic data and simulation settings.
+
+        This constructor sets up the inversion framework by storing the input
+        data and simulation configuration, initializing caches, and preparing
+        the random number generator for Monte Carlo sampling.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            MultiIndex DataFrame containing acoustic measurements with required
+            columns 'sv_mean', 'nasc', and 'thickness_mean' at top level,
+            and 'frequency' as nested index level.
+        simulation_settings : Dict[str, Any]
+            Dictionary containing simulation configuration including:
+            - environment: Environmental parameters (sound speed, density)
+            - monte_carlo: Whether to use Monte Carlo initialization
+            - mc_realizations: Number of MC realizations (if monte_carlo=True)
+            - scale_parameters: Whether to scale parameters to [0,1]
+            - minimum_frequency_count: Min frequencies required for inversion
+        verbose : bool, default=True
+            Whether to print progress messages during initialization and inversion
+
+        Notes
+        -----
+        The constructor performs minimal initialization, deferring heavy
+        computation until build_scattering_model() is called. This allows
+        for efficient object creation and parameter validation.
```

Comment:
I don't think the second sentence adds information.
```suggestion
        The constructor performs minimal initialization, deferring heavy
        computation until build_scattering_model() is called.
```